code/game/objects/items/plushes.dm

```diff
@@ -7,14 +7,64 @@
 	w_class = WEIGHT_CLASS_SMALL
 	resistance_flags = FLAMMABLE
 	var/list/squeak_override //Weighted list; If you want your plush to have different squeak sounds use this
+	var/stuffed = TRUE //If the plushie has stuffing in it
+	var/obj/item/grenade/grenade //You can remove the stuffing from a plushie and add a grenade to it for *nefarious uses*
 
 /obj/item/toy/plush/Initialize()
 	. = ..()
 	AddComponent(/datum/component/squeak, squeak_override)
 
 /obj/item/toy/plush/attack_self(mob/user)
 	. = ..()
-	to_chat(user, "<span class='notice'>You pet [src]. D'awww.</span>")
+	if(stuffed || grenade)
+		to_chat(user, "<span class='notice'>You pet [src]. D'awww.</span>")
+		if(grenade && !grenade.active)
+			if(istype(grenade, /obj/item/grenade/chem_grenade))
+				var/obj/item/grenade/chem_grenade/G = grenade
+				if(G.nadeassembly) //We're activated through different methods
+					return
+			playsound(user, 'sound/weapons/armbomb.ogg', 10, TRUE)
+			addtimer(CALLBACK(grenade, /obj/item/grenade/proc/prime), grenade.det_time)
```

Comment:
These two lines should be a /grenade proc

Reply:
What do you mean by that? Should the `/obj/item/grenade` typepath be replaced with something like `grenade.proc/prime`?